src/ack_test_infra/k8s/resource.py

```diff
@@ -0,0 +1,381 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""Helper methods and structures for cleanly interacting with the K8s
+CustomResource APIs.
+"""
+
+import logging
+import base64
+from time import sleep
+from typing import Dict, Optional, Union
+from dataclasses import dataclass
+from kubernetes import config, client
+from kubernetes.client.api_client import ApiClient
+from kubernetes.client.rest import ApiException
+
+from ..resources import load_resource_file
+
+_k8s_api_client = None
+
+
+@dataclass
+class CustomResourceReference:
+    """Stores a reference to a CustomResource within the cluster.
+
+    Rather than repeating each field of the custom resource reference each time,
+    we can pass around a single class with all the fields specified.
+    """
+
+    group: str
+    version: str
+    plural: str
+    name: str
+    namespace: Optional[str] = None
+
+    def __str__(self):
+        return self.to_short_resource_string()
+
+    @property
+    def _printable_namespace(self):
+        return self.namespace or "cluster"
+
+    def to_short_resource_string(self):
+        return f"{self._printable_namespace}:{self.name}"
+
+    def to_long_resource_string(self):
+        return f"{self.plural}.{self.version}.{self.group}/{self._printable_namespace}:{self.name}"
+
+def load_resource(service_name: str,
```

Comment:
I'd love to add `create()` and `delete()` methods to the `CustomResourceReference` object above and have this function be called just `from_file` and make the calling sequence similar to this eventually:

```python
from acktest.k8s import resource

from test.e2e import SERVICE_NAME
...
my_res = resource.from_file(SERVICE_NAME, "path/to/resource.yaml", replacements)
...
my_res.create()
...
my_res.delete()
```

We could then make all the functions in this file private and hide the implementation details as much as possible from the callers.